### PR TITLE
test/lang: add basic smoketest for SQLAlchemy

### LINKED ISSUE #4913: Python SqlAlchemy client fails with 'column "typarray" does not exist'
### What version of Materialize are you using?

```
$ materialized -v

materialized v0.5.2 (58333a7e8) listening on 0.0.0.0:6875...

```

### How did you install Materialize?

* [x] Docker image
* [ ] Linux release tarball
* [ ] APT package
* [ ] macOS release tarball
* [ ] Homebrew tap
* [ ] Built from source

### What was the issue?

A basic SELECT fails, it would be nice to have a very minimal integration (at least SELECTs on basic types, to run the demos, would have a good ROI if it requires just some small modifications). 

### Is the issue reproducible? If so, please provide reproduction instructions.

```
from sqlalchemy.engine import create_engine
engine = create_engine('postgresql://localhost:6875/materialize')

engine.execute('SELECT 1')

```

```
pip freeze 
psycopg2==2.8.6
SQLAlchemy==1.3.8
```

### Please attach any applicable log files.

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/hue/build/env/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 2178, in execute
    connection = self._contextual_connect(close_with_result=True)
  File "/hue/build/env/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 2239, in _contextual_connect
    self._wrap_pool_connect(self.pool.connect, None),
  File "/hue/build/env/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 2278, in _wrap_pool_connect
    Connection._handle_dbapi_exception_noconnection(
  File "/hue/build/env/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1544, in _handle_dbapi_exception_noconnection
    util.raise_from_cause(sqlalchemy_exception, exc_info)
  File "/hue/build/env/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 398, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/hue/build/env/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 152, in reraise
    raise value.with_traceback(tb)
  File "/hue/build/env/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 2275, in _wrap_pool_connect
    return fn()
  File "/hue/build/env/lib/python3.8/site-packages/sqlalchemy/pool/base.py", line 363, in connect
    return _ConnectionFairy._checkout(self)
  File "/hue/build/env/lib/python3.8/site-packages/sqlalchemy/pool/base.py", line 760, in _checkout
    fairy = _ConnectionRecord.checkout(pool)
  File "/hue/build/env/lib/python3.8/site-packages/sqlalchemy/pool/base.py", line 492, in checkout
    rec = pool._do_get()
  File "/hue/build/env/lib/python3.8/site-packages/sqlalchemy/pool/impl.py", line 139, in _do_get
    self._dec_overflow()
  File "/hue/build/env/lib/python3.8/site-packages/sqlalchemy/util/langhelpers.py", line 68, in __exit__
    compat.reraise(exc_type, exc_value, exc_tb)
  File "/hue/build/env/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 153, in reraise
    raise value
  File "/hue/build/env/lib/python3.8/site-packages/sqlalchemy/pool/impl.py", line 136, in _do_get
    return self._create_connection()
  File "/hue/build/env/lib/python3.8/site-packages/sqlalchemy/pool/base.py", line 308, in _create_connection
    return _ConnectionRecord(self)
  File "/hue/build/env/lib/python3.8/site-packages/sqlalchemy/pool/base.py", line 437, in __init__
    self.__connect(first_connect_check=True)
  File "/hue/build/env/lib/python3.8/site-packages/sqlalchemy/pool/base.py", line 647, in __connect
    pool.dispatch.first_connect.for_modify(
  File "/hue/build/env/lib/python3.8/site-packages/sqlalchemy/event/attr.py", line 314, in exec_once_unless_exception
    self._exec_once_impl(True, *args, **kw)
  File "/hue/build/env/lib/python3.8/site-packages/sqlalchemy/event/attr.py", line 285, in _exec_once_impl
    self(*args, **kw)
  File "/hue/build/env/lib/python3.8/site-packages/sqlalchemy/event/attr.py", line 322, in __call__
    fn(*args, **kw)
  File "/hue/build/env/lib/python3.8/site-packages/sqlalchemy/engine/strategies.py", line 189, in on_connect
    do_on_connect(conn)
  File "/hue/build/env/lib/python3.8/site-packages/sqlalchemy/dialects/postgresql/psycopg2.py", line 846, in on_connect
    fn(conn)
  File "/hue/build/env/lib/python3.8/site-packages/sqlalchemy/dialects/postgresql/psycopg2.py", line 813, in on_connect
    hstore_oids = self._hstore_oids(conn)
  File "/hue/build/env/lib/python3.8/site-packages/sqlalchemy/util/langhelpers.py", line 876, in oneshot
    result = fn(self, *args, **kw)
  File "/hue/build/env/lib/python3.8/site-packages/sqlalchemy/dialects/postgresql/psycopg2.py", line 903, in _hstore_oids
    oids = extras.HstoreAdapter.get_oids(conn)
  File "/hue/build/env/lib/python3.8/site-packages/psycopg2/extras.py", line 927, in get_oids
    curs.execute("""\
sqlalchemy.exc.InternalError: (psycopg2.errors.InternalError_) column "typarray" does not exist

(Background on this error at: http://sqlalche.me/e/2j85)

```

### PR DESCRIPTION
**Pull request by benesch**
This is also an indirect test of psycopg2, which SQLAlchemy uses under
the hood.

Fix #4913.

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/4928)

### CODE REVIEW

File: test/lang/python/smoketest.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# by the Apache License, Version 2.0.
+
+import sqlalchemy
+import unittest
+
+
+class SmokeTest(unittest.TestCase):
+    def test_sqlalchemy(self):
+        engine = sqlalchemy.engine.create_engine(
+            "postgresql://materialized:6875/materialize"
```
Comment (romainr) [inline test/lang/python/smoketest.py:17]: +1

File: src/sql/src/plan/func.rs [hunk 2]
```diff
@@ -1036,6 +1036,9 @@ lazy_static! {
             "convert_from" => Scalar {
                 params!(Bytes, String) => BinaryFunc::ConvertFrom
             },
+            "current_schema" => Scalar {
+                params!() => sql_op!("current_schemas(false)[1]")
```
Comment (JLDLaughlin) [inline src/sql/src/plan/func.rs:1040]: nice!

File: test/lang/python/smoketest.py (hunk 1 above)
Comment (JLDLaughlin) [reply test/lang/python/smoketest.py:17]: agreed!

**Review by JLDLaughlin — APPROVED** [review]
looks great!

**Pull request merged**